### PR TITLE
Windows support

### LINKED ISSUE #8: Adopt pathlib drive and root structure?
I tried to fix things up on Windows, but I'm starting to wonder whether it might make sense to go with the structure of having a drive and a root property of every path, like Python's [pathlib](https://www.python.org/dev/peps/pep-0428/) does. One could then also follow the definition of an anchored path from pathlib, my sense is that to really square the design on Windows and Posix, it would be easier to just follow Python's lead on those things.

In general, I assume pathlib was kind of the model for this package?

### LINKED ISSUE #12: Fix no parents test failure on windows
```julia
ERROR: LoadError: LoadError: path.jl has no parents
Stacktrace:
 [1] parents(::FilePaths.WindowsPath) at C:\Users\appveyor\.julia\v0.6\FilePaths\src\path.jl:83
 [2] parent(::FilePaths.WindowsPath) at C:\Users\appveyor\.julia\v0.6\FilePaths\src\path.jl:60
 [3] include(::String) at .\boot.jl:232
 [4] include_from_node1(::String) at .\loading.jl:569
 [5] include(::String) at .\sysimg.jl:14
 [6] include_from_node1(::String) at .\loading.jl:569
 [7] include(::String) at .\sysimg.jl:14
 [8] process_options(::Base.JLOptions) at .\client.jl:305
 [9] _start() at .\client.jl:371
while loading C:\Users\appveyor\.julia\v0.6\FilePaths\test\path.jl, in expression starting on line 2
while loading C:\Users\appveyor\.julia\v0.6\FilePaths\test\runtests.jl, in expression starting on line 16
==============================[ ERROR: FilePaths ]==============================
```

### PR DESCRIPTION
**Pull request by davidanthoff**
Fixes #12, fixes #8.

I think this is ready now. I'm pretty sure there are still corner cases that it doesn't handle correctly, but for now tests pass and it seems easier to fix these as we go along.

I dropped julia 0.5 support, hope that is ok.

### CODE REVIEW

**Comment by davidanthoff** [comment]
I have a follow up PR to this one that fixes lots of julia 0.6 deprecations, but I'll open that once this one here is merged.

File: src/FilePaths.jl [hunk 1]
```diff
@@ -66,6 +68,7 @@ Base.next(p::AbstractPath, i::Int) = next(String(p), i)
 Base.String(path::AbstractPath) = error("`String not implemented")
 parts(path::AbstractPath) = error("`parts` not implemented.")
 root(path::AbstractPath) = error("`root` not implemented.")
+drive(path::AbstractPath) = error("`drive` not implemented.")
```
Comment (rofinn) [inline src/FilePaths.jl]: Should this default to returning an empty string? https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.drive

File: src/windows.jl [hunk 2]
```diff
@@ -1,46 +1,75 @@
...
+
+        return WindowsPath(tuple(String.(tokenized)...), l_drive, l_root)
+    else
+        tokenized = split(str, WIN_PATH_SEPARATOR)
+
+        return WindowsPath(tuple(String.(tokenized)...), "", "")
+    end
+end
+
+==(a::WindowsPath, b::WindowsPath) =
```
Comment (rofinn) [inline src/windows.jl]: I'd probably using 
```
function ==(...)
    ...
end
``` 
if we're going to split this across multiple lines.

File: test/path.jl [hunk 3]
```diff
@@ -106,22 +116,24 @@ mktmpdir() do d
...
-            @test !isexecutable(p"newfile")
-
-            @test String(mode(p"newfile")) == "-rw-r--r--"
-            chmod(p"newfile", "+x")
-            write(p"newfile", "foobar")
-            @test read(p"newfile") == "foobar"
-            chmod(p"newfile", "u=rwx")
-
-            chmod(new_path, mode(p"newfile"); recursive=true)
+            @static if is_unix()
```
Comment (rofinn) [inline test/path.jl]: It'd be nice to at least partially support `chmod` on windows similar to python, but that's probably better as a separate change.

File: src/FilePaths.jl (hunk 1 above)
Comment (davidanthoff) [reply src/FilePaths.jl]: I mainly tried to follow the convention for the other public API functions here that all had one method that throws an error for ``AbstractPath``, and then two separate methods that work for real, one for ``PosixPath`` and one for ``WindowsPath``. So there is a ``drive(path::PosixPath) = ""`` defind somewhere else.

Happy to change it, but shouldn't we change it for ``parts`` and ``root`` as well then?

File: test/path.jl (hunk 3 above)
Comment (davidanthoff) [reply test/path.jl]: Yeah, we can fix that later: https://github.com/rofinn/FilePaths.jl/issues/20

File: src/windows.jl (hunk 2 above)
Comment (davidanthoff) [reply src/windows.jl]: Done.

File: src/FilePaths.jl (hunk 1 above)
Comment (rofinn) [reply src/FilePaths.jl]: Oops, I forgot about the existing `drive(path::PosixPath) = ""` method. N/m that works then.

**Comment by rofinn** [comment]
Alright, this looks good to merge. Thanks @davidanthoff!

**Comment by davidanthoff** [comment]
Cool. Maybe tag a version? Then I can start to use it in various packages and see how it goes. I'm pretty sure this will need a couple of iterations to iron out all the issues...

**Comment by rofinn** [comment]
Sounds good. https://github.com/JuliaLang/METADATA.jl/pull/13436
